### DISCUSSION TITLE
Trying to find local TDof number of Global TDof

### DISCUSSION THREAD

**Discussion by JacobLotz** [mfem/mfem#3013] [Q&A] [answered]
I am trying to find the local TDof number of a global Tdof. 
In my `ParFiniteElementSpace` I have 108 global Tdofs which are distributed over two processes, giving a total of 126 global dofs (63 per proces). Of these global Tdofs I want to know if they are in the current proces and what their local true number is.

If I know the global Dof I can get the local TDof number using: `int GetLocalTDofNumber(int ldof) const;`

However, in order to do this, I first have to find the global Dof. I have the idea that I can do this using  `Dof_TrueDof_Matrix()` but I am not sure if this is the correct procedure or there is an easier procedure.

Does anyone have suggestion?

**Comment by v-dobrev** [comment, marked as answer]
You can check the global tdof index against the global tdof offset for the current rank which you can get from the method
```c++
   HYPRE_BigInt ParFiniteElementSpace::GetMyTDofOffset() const;
```
Basically the global tdof index is the sum of the global tdof offset plus the local tdof index. The last global tdof index on the current rank is: `my_global_tdof_offset + ParFiniteElementSpace::GetTrueVSize() - 1`.

**Reply by JacobLotz** [reply]
Thank you! This is exactly what I needed!
